src/helpers/gateway/mod.rs

```diff
@@ -59,6 +61,13 @@ impl<T: Transport> Gateway<T> {
         roles: RoleAssignment,
         transport: T,
     ) -> Self {
+        let senders = Arc::new(GatewaySenders::default());
+        let receivers = Arc::new(GatewayReceivers::default());
+        let mut handle :Option<tokio::task::JoinHandle<()>> = None;
```

Comment:
`let handle = if cfg!(debug_assertions) { Some(Self::create(...)) } else { None }`

can we consider working with `GatewaySenders` rather than `Arc<GatewaySenders>` if debug assertions are disabled?

Reply:
Sounds good

Reply:
Sure. Moved Arc to debug only